yt_dlp/extractor/cda.py

```diff
@@ -83,8 +98,92 @@ def _download_age_confirm_page(self, url, video_id, *args, **kwargs):
                 'Content-Type': content_type,
             }, **kwargs)
 
+    def _login(self, username, password):
+        token = [c for c in self._downloader.cookiejar if c.domain == 'ytdlp-internal.cda.pl' and c.name == 'token']
+        valid_until = [c for c in self._downloader.cookiejar if c.domain == 'ytdlp-internal.cda.pl' and c.name == 'valid_until']
+        if len(token) > 0 and len(valid_until) > 0 and float(valid_until[0].value) > datetime.datetime.now().timestamp() + 5:
+            return token[0].value
+
+        password_hash = base64.urlsafe_b64encode(hmac.new(
+            's01m1Oer5IANoyBXQETzSOLWXgWs01m1Oer5bMg5xrTMMxRZ9Pi4fIPeFgIVRZ9PeXL8mPfXQETZGUAN5StRZ9P'.encode('utf-8'),
+            ''.join((
+                    '{:0>2}'.format(bytes((bt & 255, )).hex())
+                    for bt
+                    in hashlib.md5(password.encode('utf-8')).digest()
+                    )).encode('utf-8'),
+            hashlib.sha256).digest(),
+        ).decode('utf-8').replace('=', '')
+
+        headers = self._BASE_API_HEADERS
+        headers.update({
+            'Authorization': self._LOGIN_REQUEST_AUTH,
+        })
+
+        token_res = self._download_json(f'{self._BASE_API_URL}/oauth/token?grant_type=password&login={username}&password={password_hash}',
+                                        None, 'Logging in', headers=headers, data=bytes(''.encode('utf-8')))
+        self._set_cookie('ytdlp-internal.cda.pl', 'token', token_res['access_token'])
+        self._set_cookie('ytdlp-internal.cda.pl', 'valid_until', str(token_res['expires_in'] + datetime.datetime.now().timestamp()))
+
+        return token_res['access_token']
+
     def _real_extract(self, url):
         video_id = self._match_id(url)
+
+        (username, password) = self._get_login_info()
+        if username is not None:
+            token = self._login(username, password)
+            return self._api_extract(video_id, token)
+        else:
+            return self._web_extract(video_id, url)
+
+    def _api_extract(self, video_id, token):
+        headers = self._BASE_API_HEADERS
+        headers.update({
+            'Authorization': f'Bearer {token}',
+        })
+
+        meta = self._download_json(
+            f'{self._BASE_API_URL}/video/{video_id}', video_id, headers=headers)['video']
+
+        if meta.get('premium') is True and meta.get('premium_free') is not True:
```

Comment:
any reason this isnt just
```suggestion
        if meta.get('premium') and not meta.get('premium_free'):
```